synapseclient/client.py

```diff
@@ -2127,39 +2127,55 @@ def _getUserCredentials(self, url, username=None, password=None):
 
     def createStorageLocationSetting(self, storage_type, **kwargs):
         """
-        Creates a stroage location based on the specified type.
+        Creates an IMMUTABLE storage location based on the specified type.
+
+        For each storage_type, the following kwargs should be specified:
+        ExternalObjectStorage: (S3-like bucket not accessed/managed by synapse)
+         - endpointUrl: endpoint URL of the S3 service (for example: 'https://s3.amazonaws.com')
+         - bucket: the name of the bucket to use
+        ExternalS3Storage: (S3-like bucket accessed/managed by synapse)
+         - bucket: the name of the bucket to use
+        ExternalStorage: (SFTP or FTP storage location not accessed/managed by synapse)
+         - url: the base URL for uploading to the external destination
+         - supportsSubfolders(optional): does the destination support creating subfolders under the base url (default: false)
+        ProxyStorage: (a proxy server to a storage not accessed/managed by Synapse)
```

Comment:
'a proxy server to storage not accessed/managed by Synapse'

Though I find it confusing that Synapse does not 'access' this storage location - how does it work then?